### PR TITLE
DOC - add link to source code

### LINKED ISSUE #250: DOC link to source code in API documentation
It would be nice if the doc could automatically point to the source code (e.g. in https://mathurinm.github.io/celer/generated/celer.datasets.fetch_ml_uci.html#celer.datasets.fetch_ml_uci) 

sklearn does it (bottom right)
image

so it should be doable easily with some `sphinx` option

### PR DESCRIPTION
**Pull request by Badr-MOUFAD**
fixes #250 

This adds links to github source code. 

Changes are inspired by scikit-learn repo.

### CODE REVIEW

**Comment by mathurinm** [comment]
Nice ! It works on almost all methods, but for some it does not e.g. : https://output.circle-artifacts.com/output/job/d218fa10-2aaf-41cd-ae85-900713ae69fb/artifacts/0/dev/generated/celer.datasets.fetch_libsvm.html#celer.datasets.fetch_libsvm

Do you know why ?

**Comment by Badr-MOUFAD** [comment]
I inspect the rendered page and it appears that the link to the source code is actually generated yet it 's not active.
That's weird !

inspect-doc-devtool

**Comment by mathurinm** [comment]
there is a \ in front of libsvm.py instead of /, could it be the reason ?

**Comment by Badr-MOUFAD** [comment]
I don't think so. It is the same for ``make_correlated_data``, however it works.

counter-examples

File: doc/_static/style.css [hunk 1]
```diff
@@ -18,9 +18,9 @@ code {
     color: #2c3e50 !important;
 }
 
-.function dt {
+/* .function dt {
```
Comment (Badr-MOUFAD) [inline doc/_static/style.css:21]: I think custom styling is the problem (as inspecting DOM revealed that links are correctly generated).
Overriding the ``dt`` tag compromises the behavior of the ``a`` tag.

Comment (mathurinm) [reply doc/_static/style.css:21]: hats off !

**Pull request merged**